### PR TITLE
added notiication feature

### PR DESCRIPTION
**Pull request by ppauljonathan**
added actioncable to show notifications to users with particular topics when question related to that topic has been posted

### CODE REVIEW

File: app/models/question.rb [hunk 1]
```diff
@@ -19,6 +20,7 @@ class Question < ApplicationRecord
   has_many_attached :files
   has_many :answers
   has_many :comments, as: :commentable
+  has_many :notifications
```
Comment (shubhamjuneja1) [inline app/models/question.rb:23]: dependent

**Review by shubhamjuneja1 — APPROVED** [review]

File: db/migrate/20230605132213_create_notifications.rb [hunk 2]
```diff
@@ -0,0 +1,12 @@
+class CreateNotifications < ActiveRecord::Migration[7.0]
+  def change
+    create_table :notifications do |t|
+      t.references :question, null: false, foreign_key: true
```
[the remark this answers is not in the document]
Comment (ppauljonathan) [reply db/migrate/20230605132213_create_notifications.rb:4]: changed to `notifiable` which is polymorphic, also added `notifiable` concern and also changed channel name from `question_channel` to `notification_channel`

File: app/models/user.rb [hunk 3]
```diff
@@ -24,6 +24,7 @@ class User < ApplicationRecord
   end
   has_many :orders
   has_many :credit_transactions
+  has_many :notifications
```
[the remark this answers is not in the document]
Comment (ppauljonathan) [reply app/models/user.rb:27]: done

File: app/models/question.rb (hunk 1 above)
Comment (ppauljonathan) [reply app/models/question.rb:23]: done

**Review by shubhamjuneja1 — APPROVED** [review]

**Pull request merged**